cli/remotebazel/remotebazel.go

```diff
@@ -125,69 +132,106 @@ type RepoConfig struct {
-	if confRemote != "" {
-		r, err := repo.Remote(confRemote)
-		if err == nil {
-			return r, nil
+	// If multiple remotes are configured, check if a remote was previously
+	// used and cached in the buildbuddy config.
+	cachedRemote, _ := storage.ReadRepoConfig(gitConfigRemoteBazelRemote)
+	if cachedRemote != "" {
+		for _, r := range remotes {
+			if r.name == cachedRemote {
+				return r, nil
+			}
 		}
-		log.Debugf("Could not find remote %q saved in config, ignoring", confRemote)
+		log.Debugf("Could not find remote %q saved in config, ignoring", cachedRemote)
 	}
 
+	// Prompt user to select a remote if there are multiple.
 	var remoteNames []string
 	for _, r := range remotes {
-		if len(r.Config().URLs) > 0 && r.Config().Name != "" {
-			remoteNames = append(remoteNames, fmt.Sprintf("%s (%s)", r.Config().Name, r.Config().URLs[0]))
-		}
-	}
-
-	if len(remoteNames) == 0 {
-		return nil, status.InvalidArgumentErrorf("invalid .git/config - no remote URLs")
+		remoteNames = append(remoteNames, fmt.Sprintf("%s (%s)", r.name, r.url))
 	}
 
 	selectedRemoteAndURL := ""
-	if len(remoteNames) == 1 {
-		selectedRemoteAndURL = remoteNames[0]
-	} else {
-		prompt := &survey.Select{
-			Message: "Select the git remote that will be used by the remote Bazel instance to fetch your repo:",
-			Options: remoteNames,
-		}
-		if err := survey.AskOne(prompt, &selectedRemoteAndURL); err != nil {
-			return nil, err
-		}
+	prompt := &survey.Select{
+		Message: "Select the git remote that will be used by the remote Bazel instance to fetch your repo:",
+		Options: remoteNames,
+	}
+	if err := survey.AskOne(prompt, &selectedRemoteAndURL); err != nil {
+		return nil, err
 	}
 
 	selectedRemote := strings.Split(selectedRemoteAndURL, " (")[0]
-	remote, err := repo.Remote(selectedRemote)
-	if err != nil {
-		return nil, err
+	for _, r := range remotes {
+		if r.name == selectedRemote {
+			err = storage.WriteRepoConfig(gitConfigRemoteBazelRemote, r.name)
+			if err != nil {
+				log.Warnf("failed to cache selected remote in .git/config: %s", err)
+			}
+			return r, nil
+		}
 	}
 
-	conf.Raw.Section(gitConfigSection).SetOption(gitConfigRemoteBazelRemote, selectedRemote)
-	if err := repo.SetConfig(conf); err != nil {
-		return nil, status.WrapError(err, "invalid .git/config")
+	return nil, status.InternalError("selected remote is not configured")
+}
+
+// parseRemote expects a string that looks like
+// "origin	git@github.com:buildbuddy-io/buildbuddy.git (fetch)"  and will parse
+// it into a `gitRemote` struct.
+func parseRemote(s string) (*gitRemote, error) {
+	parts := strings.Fields(s)
+	if len(parts) < 2 {
+		return nil, status.InvalidArgumentErrorf("invalid remote %s", s)
 	}
+	r := &gitRemote{
+		name: parts[0],
+		url:  parts[1],
+	}
+	if len(parts) >= 3 {
+		use := parts[2]
+		use = strings.TrimSuffix(use, ")")
+		use = strings.TrimPrefix(use, "(")
+		r.use = use
+	}
+
+	return r, nil
```

Comment:
FWIW, over in `gh` cli, they use a regex to parse this string https://github.com/cli/cli/blob/b133f9cce0767071c004708c83b7d934187a63d0/git/client.go#L22

Reply:
Oh nice that's cleaner - thanks